pinecone/client.go

```diff
@@ -1506,9 +1509,9 @@ func toIndex(idx *db_control.IndexModel) *Index {
 		spec.Pod = &PodSpec{
 			Environment:      idx.Spec.Pod.Environment,
 			PodType:          idx.Spec.Pod.PodType,
-			PodCount:         int32(idx.Spec.Pod.Pods),
-			Replicas:         idx.Spec.Pod.Replicas,
-			ShardCount:       idx.Spec.Pod.Shards,
+			PodCount:         derefOrDefault(idx.Spec.Pod.Pods, 0),
```

Comment:
I don't think these should ever realistically be `nil` when coming back from the server.

Reply:
I agree... maybe the default should be 1 (as that's the count if all other defaults are used)